internal/status/kv.go

```diff
@@ -48,8 +49,8 @@ func ConnectToKVStores(s events.Stream, log *logrus.Logger, opts ...kv.Option) {
 
 // WatchFirmwareInstallStatus specializes some generic NATS functionality, mainly to keep
 // the callers cleaner of the NATS-specific details.
-func WatchFirmwareInstallStatus(ctx context.Context) (nats.KeyWatcher, error) {
-	// we can restrict the keys we watch (e.g. by facility code) here by using
-	// the KV Watch function instead.
-	return firmwareInstallKV.WatchAll(nats.Context(ctx))
+func WatchFirmwareInstallStatus(ctx context.Context, facility string) (nats.KeyWatcher, error) {
```

Comment:
Could we stop this watcher once the Condition has been finalised? since we're going to be creating a lot more of these.

Reply:
The Watcher that is returned here watches the entire bucket, so this object handles all conditions coming in from the configured facility. We still only need one per instance.

Reply:
Ah okay, I read it correctly now - for some reason I read `keyStr` to be `facility.ConditionID`